gflownet/envs/crystals.py

```diff
@@ -332,17 +334,31 @@ def step(self, action: Tuple[int, int]) -> Tuple[List[int], Tuple[int, int], boo
             return self.state, action, valid
         # If action is eos, then perform eos
         else:
-            if self.get_mask_invalid_actions()[-1]:
-                valid = False
+            if self._can_produce_neutral_charge():
+                self.done = True
+                valid = True
+                self.n_actions += 1
             else:
-                if self._can_produce_neutral_charge():
-                    self.done = True
-                    valid = True
-                    self.n_actions += 1
-                else:
-                    valid = False
+                valid = False
             return self.state, self.eos, valid
 
+    def set_state(self, state: List, done: Optional[bool] = False):
+        """
+        Sets the state and done. If done is True but incompatible with state (space
+        group is missing), then force done False and print warning.
+        """
+        state_elem = [self.idx2elem[i] for i, e in enumerate(state) if e > 0]
+        n_state_atoms = sum(state)
+        if done == True and not self._can_eos(state, state_elem, n_state_atoms):
```

Comment:
Cosmetic again, but the proper `True` check in Python is `if var is True` not `if var == True`

Reply:
Sure, changed [here](https://github.com/alexhernandezgarcia/gflownet/pull/138/commits/1c85f2be20ac63e8903c571926f92bb1247b9ac2).

Reply:
(this is part of another PR merged here though)